### ISSUE TITLE
[5.1] Dialog scrolling

### ISSUE THREAD

**Issue by C-Lodder** [joomla/joomla-cms#43076]
### Steps to reproduce the issue

In your administrator dashboard, click on the "Add module to your dashboard", which will open a dialog.
When the dialog is open, you are still able to scroll on the main page (behind the dialog).

### Expected result

When the dialog is open, I'd expect scrolling to be disabled for the main page.

![image](https://github.com/joomla/joomla-cms/assets/2019801/2d1c3449-6ed8-4490-90b6-efe5b75924f2)

**Comment by Fedik** [comment]
No. Does the dialog scrolls or moves when you scroll the page? No, then no point :wink:

**Comment by C-Lodder** [comment]
@Fedik No, but when scrolling has reached the bottom within the dialog, it starts scrolling the main page. This is because the main page is still scrollable.

Body scrolling is disabled with modals are opened with all major frameworks. Only HTML dialogs don't do this by default because because they have poor usability and accessibility.

**Comment by Fedik** [comment]
I do not see it as an issue, maybe minor. 
Overal accesebility of native dialog much better than most of major freamworks modals (especialy with focus handling).


> The APA WG supports this proposal; it would improve accessibility for people who are experiencing a range of vision and cognition barriers.

With jumpy page, it does not make the accessibility better isn't it?
I mean, when you do "overflow:hidden" on the page, then much more things can go wrong: position:sticky stops working, browser removes "scroll bar", in the result the page content jumps around.

> One caveat I can think of with this idea is that if an author makes a dialog that is too large for the current viewport, then the underlying document will need to be able to scroll so as to allow someone to view all the content of the dialog.

From the linked post.

I would keep native browser behavior here.

**Comment by Fedik** [comment]
hm, maybe if it would be possible to prevent scroll event on parent window, then it would be possible, to mitigate it somehow.

**Comment by Ruud68** [comment]
Just 'bumping' this one here. Been rewriting a lot of BS modals into the new dialogs. When you have a long page with a lot of links to dialogs, then this is actually a user interface 'issue'. When scrolling the dialog is at the bottom, the background automatically starts scrolling: when closing the dialog the user is 'lost' as that is not where he started / clicked the link to open the dialog: he needs to scroll back to find where he left off.

The old (and yes I am aware of the phrase that in the past everything was better :)) BS modals didn't have this background scrolling enabled: works IMO better / user friendlier.

just my 2cts.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43076">issues.joomla.org/tracker/joomla-cms/43076</a>.</sub>

**Comment by Fedik** [comment]
For reference:
- https://github.com/w3c/csswg-drafts/issues/7134
- [Vertical Scroll Policy](https://github.com/w3c/webappsec-permissions-policy/blob/main/policies/vertical_scroll.md)
